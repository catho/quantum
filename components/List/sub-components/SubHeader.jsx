import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseFontSize } from '../../shared/theme';

const Description = styled.span`
  ${{
    theme: {
      baseFontSize,
    },
  }}
  font-size: ${baseFontSize * 0.75}px;
`;

Description.displayName = 'SubHeader';

const SubHeader = ({ children, theme }) => (
  <Description theme={theme}>{children}</Description>
);

SubHeader.defaultProps = {
  theme: {
    baseFontSize,
  },
};

SubHeader.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  theme: PropTypes.shape({
    baseFontSize: PropTypes.number,
  }),
};

SubHeader.displayName = 'List.SubHeader';

export default SubHeader;
