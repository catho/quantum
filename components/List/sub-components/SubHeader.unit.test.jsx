import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SubHeader from './SubHeader';

describe('<SubHeader />', () => {
  it('Should match the snapshot', () => {
    const wrapper = mount(<SubHeader>SubHeader</SubHeader>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
