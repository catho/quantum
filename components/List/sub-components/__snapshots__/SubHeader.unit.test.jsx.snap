// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubHeader /> Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List.SubHeader>
    SubHeader
  </List.SubHeader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "SubHeader",
    },
    "ref": null,
    "rendered": "SubHeader",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "SubHeader__Description-gatzne-0",
        "isStatic": true,
        "rules": Array [
          "font-size:12px;",
        ],
      },
      "displayName": "SubHeader",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "SubHeader__Description-gatzne-0",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "SubHeader",
      },
      "ref": null,
      "rendered": "SubHeader",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "SubHeader__Description-gatzne-0",
          "isStatic": true,
          "rules": Array [
            "font-size:12px;",
          ],
        },
        "displayName": "SubHeader",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "SubHeader__Description-gatzne-0",
        "target": "span",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
